Problem 4:

rrugamba opened the pull request:
1. Test file (testing.cs)
2. Class Definition (Number(1).cs)
3. Class main (Program(2).cs)

dougbradbury commented on Testing.cs:
```diff
@@ -0,0 +1,55 @@
+            int num1 = 91;
+            int num2 = 99;
+            long pdt = number.ProductOfTwo2DigitNumbers(num1, num2);
+            Assert.AreEqual(true, number.CheckPalindromic(pdt));
```
There is no negative test case, which tells me that this solution violated rule #3: Only write enough production code as is sufficient to pass the failing test.  This test could be passed by returning a hard-coded true.

dougbradbury commented on Testing.cs:
```diff
@@ -0,0 +1,55 @@
+        {
+            int num1 = 991;
+            int num2 = 999;
+            Assert.AreEqual(990009, number.ProductOfTwo3DigitNumbers(num1, num2));
```
These two functions are the same and have nothing to do with how many digits are in the number.

dougbradbury commented on Number (1).cs:
```diff
@@ -0,0 +1,79 @@
+            int itrEnd = lengthOfnum-1;
+            if (lengthOfnum % 2 == 0)
+            {
+                //test palindromic number (even length)[910019]
```
A comment like this is a good indication that a function should be extracted.

rrugamba replied:
I added new files that have more functions, addressing all the reviews

dougbradbury commented on Number (1).cs:
```diff
@@ -0,0 +1,79 @@
+                while (str[itrBegin] == str[itrEnd])
+                {
+                    if (itrBegin == ((lengthOfnum / 2) - 1) && itrEnd == (lengthOfnum / 2))
+                        return true;
```
This function gets really deeply nested. Another sign that a function should be extracted.

rrugamba replied:
added new function that breaks the nested-ness in my previous funciton:
new file has the changes

dougbradbury commented on Testing.cs:
```diff
@@ -0,0 +1,55 @@
+        [TestInitialize]
+        public void Iniatilize()
+        {
+            number = new Number();
```
Number is not a very meaningful class name.

rrugamba commented:
Yes i changed and started from scratch, just forgot to delete the old files that are a bit confusing!

just deleted them. doublel check on the new files.
